### ISSUE TITLE
Migrate existing projects to use latest project metadata schema version 15.0.0

### ISSUE THREAD

**Issue by aaclan-ebi** [ebi-ait/dcp-ingest-central#445]
Track the work needed for migration of project metadata of all existing projects in Ingest to use the latest project schema version: update the `describedBy` to `15.0.0` version and update the ff fields

1. **estimated_cell_count** - optional property introduced introduced in `14.3.0`, source the value from existing `cellCount `(outside project metadata) in the project

2. **official_hca_publication** - required property introduced in `15.0.0`, This would need wrangler input on which projects’ `official_hca_publication` should be set to true

I think it makes sense to do the update of both fields in one go, instead of upgrading to `14.3.0` for `estimated_cell_count` then 15.0.0 for official_hca_publication. Migrating all the projects to a newer version will update ~100+ submission envelopes and exporting might take long for each change. We would have to notify Terra to just import them in one go as well.

Related to ebi-ait/dcp-ingest-central#425 (Update UI project form for estimated_cell_count)

**Comment by aaclan-ebi** [comment]
@Wkt8 /@ESapenaVentura  Do you have a list of projects which are considered official HCA publications? Could you list them here? Thanks!

**Comment by ESapenaVentura** [comment]
@aaclan-ebi the dataset tracking sheet has an official_hca_publication field!

**Comment by ESapenaVentura** [comment]
is called `hca_pub`

**Comment by aaclan-ebi** [comment]
Ah, cool, Thanks @ESapenaVentura !

**Comment by aaclan-ebi** [comment]
For this task, I need to do some testing in the dev environment to predict how long exporting will take given 100 submission envelopes.

@amnonkhen @gabsie if it's proven to take too long to export the project metadata changes given the current behaviour of the exporter and it's not acceptable, we would need to do some changes to optimise the exporter. I just want to note that this ticket may be at risk to be completed for this sprint.

**Comment by aaclan-ebi** [comment]
I will do testing in dev env to predict exporter performance for 100 submissions today.

**Comment by aaclan-ebi** [comment]
Slack thread for testing project metadata migration with DCP downstream : https://embl-ebi-ait.slack.com/archives/C01360XN04S/p1631635383071300

**Comment by aaclan-ebi** [comment]
I've picked 10 projects for testing and exported them. 

```
project uuids:
df88f39f-01a8-4b5b-92f4-3177d6c0f242
c5f46615-68de-4cf4-bbc2-a0ae10f08243
51f02950-ee25-4f4b-8d07-59aa99bb3498
d3ac7c1b-5302-4804-b611-dad9f89c049d
b963bd4b-4bc1-4404-8425-69d74bc636b8
71436067-ac41-4ace-be1b-2fbcc2cb02fa
05657a59-9f9d-4bb9-b77b-24be13aa5cea
ce33dde2-382d-448c-b6ac-bfb424644f23
1c6a960d-52ac-44ea-b728-a59c7ab9dc8e
520afa10-f9d2-4e93-ab7a-26c4c863ce18

staging areas:
gs://broad-dsp-monster-hca-dev-ebi-staging/dev/df88f39f-01a8-4b5b-92f4-3177d6c0f242
gs://broad-dsp-monster-hca-dev-ebi-staging/dev/c5f46615-68de-4cf4-bbc2-a0ae10f08243
gs://broad-dsp-monster-hca-dev-ebi-staging/dev/51f02950-ee25-4f4b-8d07-59aa99bb3498
gs://broad-dsp-monster-hca-dev-ebi-staging/dev/d3ac7c1b-5302-4804-b611-dad9f89c049d
gs://broad-dsp-monster-hca-dev-ebi-staging/dev/b963bd4b-4bc1-4404-8425-69d74bc636b8
gs://broad-dsp-monster-hca-dev-ebi-staging/dev/71436067-ac41-4ace-be1b-2fbcc2cb02fa
gs://broad-dsp-monster-hca-dev-ebi-staging/dev/05657a59-9f9d-4bb9-b77b-24be13aa5cea
gs://broad-dsp-monster-hca-dev-ebi-staging/dev/ce33dde2-382d-448c-b6ac-bfb424644f23
gs://broad-dsp-monster-hca-dev-ebi-staging/dev/1c6a960d-52ac-44ea-b728-a59c7ab9dc8e
gs://broad-dsp-monster-hca-dev-ebi-staging/dev/520afa10-f9d2-4e93-ab7a-26c4c863ce18
```

I notified the Data Import team of these 10 projects in the dcp2 thread https://embl-ebi-ait.slack.com/archives/C01360XN04S/p1632166712089600?thread_ts=1631635383.071300&cid=C01360XN04S

There are some issues I noticed:
1. Some submissions couldn't be set back to Valid because they're either Completed or Exported. I need to think of a way to simplify the manual steps needed to set them back to Valid.
2. 7 projects failed because of HTTP bad request error from Ingest Core, which still needs to be investigated, I'll paste it here.

From the standup, we realised there's also an issue that may affect the migration of DCP1 datasets. It should be possible to update metadata files except for (files which corresponding has data files). Another issue is the addition of provenance fields by the exporter. Some old datasets may be using the very old metadata schema versions and when they get reexported, the provenance fields are being added and this makes them invalid. See related DCP1 issue: ebi-ait/dcp-ingest-central#376

**Comment by aaclan-ebi** [comment]
Reexported the 10 test projects and copied the data files for testing with downstream DCP. Please see thread https://embl-ebi-ait.slack.com/archives/C01360XN04S/p1632326054101500?thread_ts=1631635383.071300&cid=C01360XN04S

**Comment by amnonkhen** [comment]
@aaclan-ebi to continue a few testing runs and publish results.

**Comment by ofanobilbao** [comment]
@aaclan-ebi did more testing on Friday. She needs to look into the logs

**Comment by aaclan-ebi** [comment]
Based on the performance testing, the estimated overall submission time for all submissions should be ~2days with 10exporter instances running which still seems acceptable.

Performance testing data: https://docs.google.com/spreadsheets/d/1S6aWcdBr3U2cWBGDQzw8V-aoMNccXIuKT2nxLP_xVNc/edit#gid=789946621

I believe we can proceed with the reexporting, hold them in the staging area until we get a heads-up from DCP downstream that we're good to import them to the Terra dataset. I will run the migration script and trigger the reexporting in batch. 

I'll follow the schedule:

~29th Sep (Wed)~ ~30th Sep (Thu)~ Batch1
Non-DCP1 datasets which have relative quick export time <=~20min each

~30th Sep (Thu)~ ~1st Oct (Fri)~ Batch 2
Non-DCP1 datasets which have a relatively long export time

~1st Oct (Fri)~ ~4th Oct (Mon)~ Batch 3
All DCP1 datasets

**Comment by aaclan-ebi** [comment]
The only needed on the exporter, for now, would be the change to not export provenance fields for metadata which has an older schema version that is not supporting it ebi-ait/dcp-ingest-central#376

I'll work on that and get it ready by Friday before I run DCP1 datasets.

**Comment by ofanobilbao** [comment]
Alegria triggered some submissions to be exported on Friday. Two groups have successfully exported over the weekend. She has wrote a PR for the third group, which is DCP1 datasets. Needs approval.

Some datasets on groups 1 and 2 have failed. But in general it has been successful.

Alegria to talk to Import team for the "good to go" for the import.

**Comment by jacobwindsor** [comment]
Please merge this once done https://github.com/ebi-ait/ingest-core/pull/82

**Comment by aaclan-ebi** [comment]
**BATCH 1**

**74 projects were successfully re-exported.**

<details>
project e07897dc-c25e-453d-b04c-27ca33adb630 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5e998380e952846aff4eb813 is exported
project a815c84b-8999-433f-958e-422c0720e00d submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5ee8d88f5e315a423810f7b3 is exported
project 38449aea-70b5-40db-84b3-1e08f32efe34 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5efc8a020b7b3250a575c2f2 is exported
project 455b46e6-d8ea-4611-861e-de720a562ada submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5f0f273132d8c765c3fdfa2f is exported
project c1810dbc-16d2-45c3-b45e-3e675f88d87b submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5f5f9ff5db657458f1bdc3b6 is exported
project 2086eb05-10b9-432b-b7f0-169ccc49d270 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5f293ac34f36d156a8bd93ea is exported
project 7027adc6-c9c9-46f3-84ee-9badc3a4f53b submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5f293d0e4f36d156a8bd9429 is exported
project 95f07e6e-6a73-4e1b-a880-c83996b3aa5c submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5f450cc0305bde342e26fab7 is exported
project c41dffbf-ad83-447c-a0e1-13e689d9b258 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5f3d32fd0ebe6d1e58763dea is exported
project 42d4f8d4-5422-4b78-adae-e7c3c2ef511c submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5f3d48230ebe6d1e58763f7a is exported
project 2ef3655a-973d-4d69-9b41-21fa4041eed7 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5f43b4de305bde342e26f7a5 is exported
project f48e7c39-cc67-4055-9d79-bc437892840c submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5f451baa305bde342e26fb5e is exported
project efea6426-510a-4b60-9a19-277e52bfa815 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5f574f9cdb657458f1bdae64 is exported
project 3089d311-f9ed-44dd-bb10-397059bad4dc submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/602e966c91708e06ec032969 is exported
project 1cd1f41f-f81a-486b-a05b-66ec60f81dcf submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5fcf497dcb48a221817bfe2d is exported
project 996120f9-e84f-409f-a01e-732ab58ca8b9 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5fcf5106cb48a221817bfea5 is exported
project 6c040a93-8cf8-4fd5-98de-2297eb07e9f6 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/601c0432ac1792031eee3f43 is exported
project 559bb888-7829-41f2-ace5-2c05c7eb81e9 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5fd74df9cb48a221817c14cd is exported
project 1c6a960d-52ac-44ea-b728-a59c7ab9dc8e submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5ffed7fa8ec45c6438fd8fb6 is exported
project dbd836cf-bfc2-41f0-9834-41cc6c0b235a submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60ca198ce259f0766126237e is exported
project dc1a41f6-9e09-42a6-959e-3be23db6da56 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60d9dad4c78d1b6cc5ca5c91 is exported
project 0d4b87ea-6e9e-4569-82e4-1343e0e3259f submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/606d3c33dd9aab1232d148e0 is exported
project 41fb1734-a121-4616-95c7-3b732c9433c7 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/6006d0c4c9762f5f0de9f74f is exported
project d3a4ceac-4d66-4984-9704-2570c0647a56 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/600811e5c9762f5f0de9fb9f is exported
project 16ed4ad8-7319-46b2-8859-6fe1c1d73a82 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60533bc391708e06ec077930 is exported
project ce7b12ba-664f-4f79-8fc7-3de6b1892183 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/6019344eac1792031eedfefd is exported
project 9d97f01f-9313-416e-9b07-560f048b2350 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/6025927dfa2a4e443f2808a0 is exported
project 7b947aa2-43a7-4082-afff-222a3e3a4635 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60ae5645447d4728e1979612 is exported
project e5d45579-1f5b-48c3-b568-320d93e7ca72 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60632afcdd9aab1232d0cd7e is exported
project a39728aa-70a0-4201-b0a2-81b7badf3e71 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/6050248a91708e06ec074ebe is exported
project 07073c12-8006-4710-a00b-23abdb814904 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/604b65a391708e06ec05ea73 is exported
project 2a64db43-1b55-4639-aabb-8dba0145689d submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/604fcfb491708e06ec072386 is exported
project 4bec484d-ca7a-47b4-8d48-8830e06ad6db submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/604ff08391708e06ec074106 is exported
project 842605c7-375a-47c5-9e2c-a71c2c00fcad submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/605209fe91708e06ec075d55 is exported
project 0fd8f918-62d6-4b8b-ac35-4c53dd601f71 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60549565301e6d46db75466a is exported
project 63b5b6c1-bbcd-487d-8c2e-0095150c1ecd submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/605b833d301e6d46db7574f6 is exported
project 08b794a0-519c-4516-b184-c583746254c5 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/606353bddd9aab1232d0cde2 is exported
project 5ee710d7-e2d5-4fe2-818d-15f5e31dae32 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60675c97dd9aab1232d0e00e is exported
project 24c654a5-caa5-440a-8f02-582921f2db4a submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/6075bee0dd9aab1232d22d7d is exported
project bd400331-54b9-4fcc-bff6-6bb8b079ee1f submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/6099c0c1dd9aab1232d47fed is exported
project a96b71c0-78a7-42d1-88ce-83c78925cfeb submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/608096dddd9aab1232d3087a is exported
project 0c09fade-e079-4fde-8e60-6725b8c1d84b submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/6082038add9aab1232d33d0a is exported
project 4037007b-0eff-4e6d-b7bd-8dd8eec80143 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60875440dd9aab1232d3cde3 is exported
project 955dfc2c-a8c6-4d04-aa4d-907610545d11 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/6098fb99dd9aab1232d47990 is exported
project 1ad6cee6-53a5-4804-a870-e5bd991186e0 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/609c57b0901b6d17e5f25cd4 is exported
project 53c53cd4-8127-4e12-bc7f-8fe1610a715c submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/609da4cf901b6d17e5f260ff is exported
project 962bd805-eb89-4c54-bad2-008e497d1307 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60a7ed7561fb7f28897d81ca is exported
project d3ac7c1b-5302-4804-b611-dad9f89c049d submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60abd22d61fb7f28897dd5ff is exported
project 71eb5f6d-cee0-4297-b503-b1125909b8c7 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60e564ccd5d575160aaf63a6 is exported
project c715cd2f-dc7c-44a6-9cd5-b6a6d9f075ae submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60f1c223d5d575160aafb5d3 is exported
project 51f02950-ee25-4f4b-8d07-59aa99bb3498 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60cb1648e259f076612626a3 is exported
project 71436067-ac41-4ace-be1b-2fbcc2cb02fa submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60cbde74e259f07661262d6d is exported
project d012d476-8f8c-4ff3-89d6-ebbe22c1b5c1 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60d0f0c6e259f07661263ff2 is exported
project df88f39f-01a8-4b5b-92f4-3177d6c0f242 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60d5c1ebd20b7e03a5b3807f is exported
project 376a7f55-b876-4f60-9cf3-ed7bc83d5415 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60e4957ed5d575160aaf501f is exported
project 414acced-eba0-440f-b721-befbc5642bef submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60ef0fd5d5d575160aafaf1c is exported
project 3cfcdff5-dee1-4a7b-a591-c09c6e850b11 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60f1c876d5d575160aafb741 is exported
project 14a26f57-9117-42c2-a613-891e173899cf submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60f72d23d5d575160aafc60b is exported
project 8bd2e5f6-9453-4b9b-9c56-59e3a40dc87e submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60f88196d5d575160aafca05 is exported
project dbcd4b1d-31bd-4eb5-94e1-50e8706fa192 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60fb5cfead68e2545d23b2a3 is exported
project 8787c238-89ef-4636-a57d-3167e8b54a80 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60feb080ad68e2545d23bd6f is exported
project 34cba5e9-ecb1-4d81-bf08-48987cd63073 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60ff4d3aad68e2545d23c034 is exported
project e57dc176-ab98-446b-90c2-89e0842152fd submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/610bccb8ad68e2545d23e713 is exported
project b9484e4e-dc40-4e38-9b85-4cecf5b8c068 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/610c6696ad68e2545d23ebde is exported
project 56e73ccb-7ae9-4fae-a738-acfb69936d7a submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/610d9448ad68e2545d23f0ad is exported
project 8dacb243-e918-4bd2-bb9a-aac6dc424161 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/61115ffead68e2545d23f9b5 is exported
project da2747fa-2921-42e0-afd4-39ef57b2b88b submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/6114545cad68e2545d240048 is exported
project dd7f2436-0c56-4709-bd17-e526bba4cc15 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/6116d45cad68e2545d240a69 is exported
project 8999b456-6fa6-438b-ab17-b62b1d8ec0c3 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/611fdf02ad68e2545d242762 is exported
project 9bab0f03-a725-4a13-9ab1-196e46cd80ed submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/611bf9ccad68e2545d241d77 is exported
project 4af795f7-3e1d-4341-b867-4ac0982b9efd submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/611c1e73ad68e2545d241e2b is exported
project 79b13a2a-9ca1-42a4-97bd-70208a11bea6 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/612012b9ad68e2545d2427be is exported
project b7259878-436c-4274-bfff-ca76f4cb7892 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/61237254ad68e2545d242e37 is exported
project d7845650-f6b1-4b1c-b2fe-c0795416ba7b submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/61238edead68e2545d243df6 is exported

</details>

** 9 failed to be exported:**

project 5116c081-8be7-49c5-8ce0-73b887328aa9 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60382e7c91708e06ec051417 is invalid

project bbec9297-3f64-4453-bb27-d3cdee53cd75 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60c9f179e259f07661261df4 is invalid

project 115da023-665e-4e69-a25b-860bb89bb2a5 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60a58bfd46c1727e2e8573ba is invalid

project e77ba921-bacb-4f47-9c35-b55155e28296 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/604fe02b91708e06ec0723fa is invalid

project b3938158-4e8d-4fdb-9e13-9e94270dde16 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/602cdce0fa2a4e443f28706f is invalid

~project 0b2cbf1a-3f40-4e01-a99d-901fa6091ee5 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5e623fb4f02bb753344c6e57 is already valid~
(This project is not intended to be exported, I'll reset back to Valid and removed the exported files in Terra staging area)

project a72bed96-a4f7-401a-a67a-17b8a7357faf submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5ebaad91e952846aff4f034f is invalid

project fc55bd4a-8694-4a28-9a35-59a685bda323 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5ee74a855e315a423810edf0 is archived

~project 453d7ee2-319f-496c-9862-99d397870b63 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5e7dca04e952846aff4e7992 is exporting~
(This project is not intended to be exported, I'll reset back to Valid and removed the exported files in Terra staging area)

**2 failed to be patched successfully, shouldn't have been reexported**
project 5b5f05b7-2482-468d-b76d-8f68c04a7a47 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5f2400664f36d156a8bd85db is exported

project e579d0f4-18ef-4d42-90fa-4d2afb60a862 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5fad45d3706b4c65fc988d77 is exported


**BATCH 2**

**38 projects were re-exported successfully:**
<details>
project c1a9a93d-d9de-4e65-9619-a9cec1052eaa submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5f0f1d5432d8c765c3fdf76c is exported
project 83f5188e-3bf7-4956-9544-cea4f8997756 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5f2bea1f4f36d156a8be1a44 is exported
project f2fe82f0-4454-4d84-b416-a885f3121e59 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5f3514db4f36d156a8be592c is exported
project 379ed69e-be05-48bc-af5e-a7fc589709bf submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60b9d0dd20647123c5487d0c is exported
project ce33dde2-382d-448c-b6ac-bfb424644f23 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/602d5ca8fa2a4e443f2873cf is exported
project 23587fb3-1a4a-4f58-ad74-cc9a4cb4c254 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/6042771f91708e06ec05374c is exported
project 2d846095-8a33-4f3c-97d4-585bafac13b4 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/604beb0091708e06ec0694fc is exported
project 520afa10-f9d2-4e93-ab7a-26c4c863ce18 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/604ff5ba91708e06ec0741ed is exported
project 1ce3b3dc-02f2-44a8-96da-d6d107b27a76 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60538a08301e6d46db753999 is exported
project 0c3b7785-f74d-4091-8616-a68757e4c2a8 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/605bcdf1301e6d46db7578d7 is exported
project 05657a59-9f9d-4bb9-b77b-24be13aa5cea submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60637913dd9aab1232d0cf0b is exported
project ccd1f1ba-74ce-469b-9fc9-f6faea623358 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60677f91dd9aab1232d0ef83 is exported
project eaefa1b6-dae1-4414-953b-17b0427d061e submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/606b9b58dd9aab1232d10374 is exported
project 05be4f37-4506-429b-b112-506444507d62 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/606ce932dd9aab1232d13531 is exported
project 3a694703-3084-4ece-9abe-d935fd5f6748 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/6070c27bdd9aab1232d15502 is exported
project e8808cc8-4ca0-4096-80f2-bba73600cba6 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60831db8dd9aab1232d354a4 is exported
project 03c6fce7-789e-4e78-a27a-664d562bb738 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/6075df1fdd9aab1232d24692 is exported
project d2111fac-3fc4-4f42-9b6d-32cd6a828267 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/607745c0dd9aab1232d28b15 is exported
project b51f49b4-0d2e-4cbd-bbd5-04cd171fc2fa submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60785e79dd9aab1232d2a704 is exported
project 52b29aa4-c8d6-42b4-807a-b35be94469ca submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/607f3bb6dd9aab1232d2f258 is exported
project c6ad8f9b-d26a-4811-b2ba-93d487978446 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/6081b77add9aab1232d31f17 is exported
project 6072616c-8794-4b20-8f52-fb15992ea5a4 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/6081f3c1dd9aab1232d332e2 is exported
project 2a72a4e5-66b2-405a-bb7c-1e463e8febb0 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60873651dd9aab1232d3a037 is exported
project 60ea42e1-af49-42f5-8164-d641fdb696bc submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60a54363901b6d17e5f3d322 is exported
project 946c5add-47d1-402a-97bb-a5af97e8bce7 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/608996aedd9aab1232d3d7f1 is exported
project 52d10a60-c8d1-4d06-8a5e-af0d5c0d5034 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/6089e704dd9aab1232d3dcfa is exported
project 248fcf03-16c6-4a41-b6cc-aad4d894ca42 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/609e7b4e901b6d17e5f2fa81 is exported
project b963bd4b-4bc1-4404-8425-69d74bc636b8 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60a6488461fb7f28897d7c38 is exported
project 0792db34-8047-4e62-802c-9177c9cd8e28 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60a8415f61fb7f28897da476 is exported
project 99101928-d9b1-4aaf-b759-e97958ac7403 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60a84ff161fb7f28897dba48 is exported
project 7c75f07c-608d-4c4a-a1b7-b13d11c0ad31 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/6102c778ad68e2545d23cde3 is exported
project 67a3de09-45b9-49c3-a068-ff4665daa50e submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60c1df2c6c83615dd8944876 is exported
project a9301beb-e9fa-42fe-b75c-84e8a460c733 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/614866b997ebaa5cf3527803 is exported
project 3e329187-a9c4-48ec-90e3-cc45f7c2311c submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60d4e1cbd20b7e03a5b35f9e is exported
project e77fed30-959d-4fad-bc15-a0a5a85c21d2 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60e4d3c7d5d575160aaf5d6a is exported
project e526d91d-cf3a-44cb-80c5-fd7676b55a1d submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60ecb77bd5d575160aaf9d2b is exported
project c5f46615-68de-4cf4-bbc2-a0ae10f08243 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60ed7ecfd5d575160aafa4c1 is exported
project 20f37aaf-caa1-40e6-9123-be6ce8feb2d6 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/61238513ad68e2545d24320e is exported
</details>

**3 projects failed to be re-exported:**

project f0bec3b9-db67-43c6-90cf-f6d33f1d6803 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/60d0caf4e259f07661263d60 is invalid

project 0ccd30ef-6a23-4e68-b5a3-3d9ca511b5ed submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/611af67fad68e2545d241451 is invalid

project ad98d3cd-26fb-4ee3-99c9-8a2ab085e737 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/5f1b0e98fe9c934c8b835c80 is archived

**2 projects shouldn't have been reexported yet, they failed to be patched with correct metadata update:**

project ccef38d7-aa92-4010-9621-c4c7b1182647 submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/6038015191708e06ec04ab1b is exported

project 7adede6a-0ab7-45e6-9b67-ffe7466bec1f submission https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/604677e191708e06ec05b095 is exported

**Comment by aaclan-ebi** [comment]
Noting the 16 project metadata failed to be patched:

```
project 8c3c290d-dfff-4553-8868-54ce45f4ba7f was not patched
submission count 1
submission count 504 Server Error: GATEWAY_TIMEOUT for url: https://api.ingest.archive.data.humancellatlas.org/projects/5cda9df1d96dad0008579758

project 005d611a-14d5-4fbf-846e-571a1f874f70 was not patched
submission count 1
submission count 400 Client Error:  for url: https://api.ingest.archive.data.humancellatlas.org/projects/5cdbdd82d96dad0008592f2a

project a29952d9-925e-40f4-8a1c-274f118f1f51 was not patched
submission count 1
submission count 400 Client Error:  for url: https://api.ingest.archive.data.humancellatlas.org/projects/5cdd1fd5d96dad00085af1b0

project 027c51c6-0719-469f-a7f5-640fe57cbece was not patched
submission count 1
submission count 400 Client Error:  for url: https://api.ingest.archive.data.humancellatlas.org/projects/5cdd2103d96dad00085b107d

project 9c20a245-f2c0-43ae-82c9-2232ec6b594f was not patched
submission count 1
submission count 400 Client Error:  for url: https://api.ingest.archive.data.humancellatlas.org/projects/5d848dffbddf5d0008992fd7

project 5b5f05b7-2482-468d-b76d-8f68c04a7a47 was not patched
submission count 2
submission count 400 Client Error:  for url: https://api.ingest.archive.data.humancellatlas.org/projects/5f23dfe84f36d156a8bd846f

project e579d0f4-18ef-4d42-90fa-4d2afb60a862 was not patched
submission count 1
submission count 400 Client Error:  for url: https://api.ingest.archive.data.humancellatlas.org/projects/5fac15fa706b4c65fc988c08

project f71d4ab7-8115-4208-83b6-bf03dec07b74 was not patched
submission count 0
submission count 500 Server Error:  for url: https://api.ingest.archive.data.humancellatlas.org/projects/60070d41c9762f5f0de9fa5f

project ccef38d7-aa92-4010-9621-c4c7b1182647 was not patched
submission count 1
submission count 500 Server Error:  for url: https://api.ingest.archive.data.humancellatlas.org/projects/6038015c91708e06ec04ab1d

project 5116c081-8be7-49c5-8ce0-73b887328aa9 was not patched
submission count 1
submission count 500 Server Error:  for url: https://api.ingest.archive.data.humancellatlas.org/projects/60382e7f91708e06ec051419

project 7adede6a-0ab7-45e6-9b67-ffe7466bec1f was not patched
submission count 1
submission count 500 Server Error:  for url: https://api.ingest.archive.data.humancellatlas.org/projects/604677ea91708e06ec05b097

project 31e4f51b-593f-441a-9b3c-514b031bdf23 was not patched
submission count 0
submission count 500 Server Error:  for url: https://api.ingest.archive.data.humancellatlas.org/projects/60c22894602e6c6f3c849483

project 8fd1609b-cd2d-4b4d-bb96-49ae6b8ade2f was not patched
submission count 0
submission count 500 Server Error:  for url: https://api.ingest.archive.data.humancellatlas.org/projects/60c36b72ac92ec2ecbb548d0

project d04c08a5-6303-4abc-afa2-639000752601 was not patched
submission count 0
submission count 500 Server Error:  for url: https://api.ingest.archive.data.humancellatlas.org/projects/60c36b74ac92ec2ecbb548d2

project 85f04768-cdda-4b01-a28b-10177b9b7db9 was not patched
submission count 0
submission count 500 Server Error:  for url: https://api.ingest.archive.data.humancellatlas.org/projects/60c36c39ac92ec2ecbb549b0

project 821c6251-dbea-4315-8fad-4ee1d34b1fc4 was not patched
submission count 0
submission count 500 Server Error:  for url: https://api.ingest.archive.data.humancellatlas.org/projects/60c36c80ac92ec2ecbb54a26
```

They need to be retried and investigate if errors are persistent.

**Comment by ESapenaVentura** [comment]
@aaclan-ebi investigated errors above, deployed changes to fix the issue with provenance fields in newer schema versions, in prod now but haven't re-exported yet

DCP-1 datasets may pose some issues, backwards compatibility broke.

**Comment by amnonkhen** [comment]
@aaclan-ebi to split ticket into several ticket for each class of problems in the export.

**Comment by amnonkhen** [comment]
@aaclan-ebi & @amnonkhen to discuss next steps

**Comment by aaclan-ebi** [comment]
I've made a spreadsheet to track the status of all project metadata that needs to be migrated https://docs.google.com/spreadsheets/d/1CaoB7_3MFuzx9ucGPeC8W0Sz1bmZ9FPk56otFJFRnlA/edit#gid=545939721

**Comment by aaclan-ebi** [comment]
I've checked the new versions of files in the prod staging area. It looks like there are 58 projects which have other updates exported aside from the project metadata. 

Noting the projects and the metadata type changes here:

```
03c6fce7-789e-4e78-a27a-664d562bb738/links
03c6fce7-789e-4e78-a27a-664d562bb738/metadata/project

05657a59-9f9d-4bb9-b77b-24be13aa5cea/links
05657a59-9f9d-4bb9-b77b-24be13aa5cea/metadata/project

05be4f37-4506-429b-b112-506444507d62/links
05be4f37-4506-429b-b112-506444507d62/metadata/project

07073c12-8006-4710-a00b-23abdb814904/links
07073c12-8006-4710-a00b-23abdb814904/metadata/project

0c09fade-e079-4fde-8e60-6725b8c1d84b/links
0c09fade-e079-4fde-8e60-6725b8c1d84b/metadata/project

0c3b7785-f74d-4091-8616-a68757e4c2a8/links
0c3b7785-f74d-4091-8616-a68757e4c2a8/metadata/project

0d4b87ea-6e9e-4569-82e4-1343e0e3259f/links
0d4b87ea-6e9e-4569-82e4-1343e0e3259f/metadata/project

0fd8f918-62d6-4b8b-ac35-4c53dd601f71/links
0fd8f918-62d6-4b8b-ac35-4c53dd601f71/metadata/project

16ed4ad8-7319-46b2-8859-6fe1c1d73a82/links
16ed4ad8-7319-46b2-8859-6fe1c1d73a82/metadata/project

1ad6cee6-53a5-4804-a870-e5bd991186e0/links
1ad6cee6-53a5-4804-a870-e5bd991186e0/metadata/project

1c6a960d-52ac-44ea-b728-a59c7ab9dc8e/links
1c6a960d-52ac-44ea-b728-a59c7ab9dc8e/metadata/project

1cd1f41f-f81a-486b-a05b-66ec60f81dcf/links
1cd1f41f-f81a-486b-a05b-66ec60f81dcf/metadata/project

1ce3b3dc-02f2-44a8-96da-d6d107b27a76/links
1ce3b3dc-02f2-44a8-96da-d6d107b27a76/metadata/project

2086eb05-10b9-432b-b7f0-169ccc49d270/metadata/library_preparation_protocol
2086eb05-10b9-432b-b7f0-169ccc49d270/metadata/project

23587fb3-1a4a-4f58-ad74-cc9a4cb4c254/links
23587fb3-1a4a-4f58-ad74-cc9a4cb4c254/metadata/project

248fcf03-16c6-4a41-b6cc-aad4d894ca42/links
248fcf03-16c6-4a41-b6cc-aad4d894ca42/metadata/project

24c654a5-caa5-440a-8f02-582921f2db4a/links
24c654a5-caa5-440a-8f02-582921f2db4a/metadata/project

2a64db43-1b55-4639-aabb-8dba0145689d/links
2a64db43-1b55-4639-aabb-8dba0145689d/metadata/project

2a72a4e5-66b2-405a-bb7c-1e463e8febb0/links
2a72a4e5-66b2-405a-bb7c-1e463e8febb0/metadata/project

3089d311-f9ed-44dd-bb10-397059bad4dc/links
3089d311-f9ed-44dd-bb10-397059bad4dc/metadata/project

3a694703-3084-4ece-9abe-d935fd5f6748/links
3a694703-3084-4ece-9abe-d935fd5f6748/metadata/project

4037007b-0eff-4e6d-b7bd-8dd8eec80143/links
4037007b-0eff-4e6d-b7bd-8dd8eec80143/metadata/project

41fb1734-a121-4616-95c7-3b732c9433c7/links
41fb1734-a121-4616-95c7-3b732c9433c7/metadata/project

453d7ee2-319f-496c-9862-99d397870b63/metadata/cell_suspension
453d7ee2-319f-496c-9862-99d397870b63/metadata/sequence_file

4bec484d-ca7a-47b4-8d48-8830e06ad6db/links
4bec484d-ca7a-47b4-8d48-8830e06ad6db/metadata/project

520afa10-f9d2-4e93-ab7a-26c4c863ce18/links
520afa10-f9d2-4e93-ab7a-26c4c863ce18/metadata/project

52b29aa4-c8d6-42b4-807a-b35be94469ca/links
52b29aa4-c8d6-42b4-807a-b35be94469ca/metadata/project

52d10a60-c8d1-4d06-8a5e-af0d5c0d5034/links
52d10a60-c8d1-4d06-8a5e-af0d5c0d5034/metadata/project

53c53cd4-8127-4e12-bc7f-8fe1610a715c/links
53c53cd4-8127-4e12-bc7f-8fe1610a715c/metadata/project

559bb888-7829-41f2-ace5-2c05c7eb81e9/links
559bb888-7829-41f2-ace5-2c05c7eb81e9/metadata/project

6072616c-8794-4b20-8f52-fb15992ea5a4/links
6072616c-8794-4b20-8f52-fb15992ea5a4/metadata/project

63b5b6c1-bbcd-487d-8c2e-0095150c1ecd/links
63b5b6c1-bbcd-487d-8c2e-0095150c1ecd/metadata/project

6c040a93-8cf8-4fd5-98de-2297eb07e9f6/links
6c040a93-8cf8-4fd5-98de-2297eb07e9f6/metadata/project

783c9952-a4ae-4106-a6ce-56f20ce27f88/data
783c9952-a4ae-4106-a6ce-56f20ce27f88/descriptor
783c9952-a4ae-4106-a6ce-56f20ce27f88/links
783c9952-a4ae-4106-a6ce-56f20ce27f88/metadata/analysis_file
783c9952-a4ae-4106-a6ce-56f20ce27f88/metadata/analysis_protocol
783c9952-a4ae-4106-a6ce-56f20ce27f88/metadata/cell_suspension
783c9952-a4ae-4106-a6ce-56f20ce27f88/metadata/donor_organism
783c9952-a4ae-4106-a6ce-56f20ce27f88/metadata/library_preparation_protocol
783c9952-a4ae-4106-a6ce-56f20ce27f88/metadata/process
783c9952-a4ae-4106-a6ce-56f20ce27f88/metadata/project
783c9952-a4ae-4106-a6ce-56f20ce27f88/metadata/sequence_file
783c9952-a4ae-4106-a6ce-56f20ce27f88/metadata/sequencing_protocol
783c9952-a4ae-4106-a6ce-56f20ce27f88/metadata/specimen_from_organism

7adede6a-0ab7-45e6-9b67-ffe7466bec1f/links

842605c7-375a-47c5-9e2c-a71c2c00fcad/links
842605c7-375a-47c5-9e2c-a71c2c00fcad/metadata/project

946c5add-47d1-402a-97bb-a5af97e8bce7/links
946c5add-47d1-402a-97bb-a5af97e8bce7/metadata/project

955dfc2c-a8c6-4d04-aa4d-907610545d11/links
955dfc2c-a8c6-4d04-aa4d-907610545d11/metadata/project

996120f9-e84f-409f-a01e-732ab58ca8b9/links
996120f9-e84f-409f-a01e-732ab58ca8b9/metadata/project

a39728aa-70a0-4201-b0a2-81b7badf3e71/links
a39728aa-70a0-4201-b0a2-81b7badf3e71/metadata/project

a815c84b-8999-433f-958e-422c0720e00d/descriptor
a815c84b-8999-433f-958e-422c0720e00d/links
a815c84b-8999-433f-958e-422c0720e00d/metadata/cell_suspension
a815c84b-8999-433f-958e-422c0720e00d/metadata/collection_protocol
a815c84b-8999-433f-958e-422c0720e00d/metadata/dissociation_protocol
a815c84b-8999-433f-958e-422c0720e00d/metadata/donor_organism
a815c84b-8999-433f-958e-422c0720e00d/metadata/library_preparation_protocol
a815c84b-8999-433f-958e-422c0720e00d/metadata/process
a815c84b-8999-433f-958e-422c0720e00d/metadata/project
a815c84b-8999-433f-958e-422c0720e00d/metadata/sequence_file
a815c84b-8999-433f-958e-422c0720e00d/metadata/sequencing_protocol
a815c84b-8999-433f-958e-422c0720e00d/metadata/specimen_from_organism

a96b71c0-78a7-42d1-88ce-83c78925cfeb/links
a96b71c0-78a7-42d1-88ce-83c78925cfeb/metadata/project

b51f49b4-0d2e-4cbd-bbd5-04cd171fc2fa/links
b51f49b4-0d2e-4cbd-bbd5-04cd171fc2fa/metadata/project

bd400331-54b9-4fcc-bff6-6bb8b079ee1f/links
bd400331-54b9-4fcc-bff6-6bb8b079ee1f/metadata/project

c1810dbc-16d2-45c3-b45e-3e675f88d87b/links
c1810dbc-16d2-45c3-b45e-3e675f88d87b/metadata/project

c6ad8f9b-d26a-4811-b2ba-93d487978446/links
c6ad8f9b-d26a-4811-b2ba-93d487978446/metadata/project

ccd1f1ba-74ce-469b-9fc9-f6faea623358/links
ccd1f1ba-74ce-469b-9fc9-f6faea623358/metadata/project

ccef38d7-aa92-4010-9621-c4c7b1182647/links

ce33dde2-382d-448c-b6ac-bfb424644f23/links
ce33dde2-382d-448c-b6ac-bfb424644f23/metadata/project
ce7b12ba-664f-4f79-8fc7-3de6b1892183/links
ce7b12ba-664f-4f79-8fc7-3de6b1892183/metadata/project

d2111fac-3fc4-4f42-9b6d-32cd6a828267/links
d2111fac-3fc4-4f42-9b6d-32cd6a828267/metadata/project

d3a4ceac-4d66-4984-9704-2570c0647a56/links
d3a4ceac-4d66-4984-9704-2570c0647a56/metadata/project

e07897dc-c25e-453d-b04c-27ca33adb630/descriptor
e07897dc-c25e-453d-b04c-27ca33adb630/links
e07897dc-c25e-453d-b04c-27ca33adb630/metadata/cell_suspension
e07897dc-c25e-453d-b04c-27ca33adb630/metadata/dissociation_protocol
e07897dc-c25e-453d-b04c-27ca33adb630/metadata/library_preparation_protocol
e07897dc-c25e-453d-b04c-27ca33adb630/metadata/process
e07897dc-c25e-453d-b04c-27ca33adb630/metadata/project
e07897dc-c25e-453d-b04c-27ca33adb630/metadata/sequence_file
e07897dc-c25e-453d-b04c-27ca33adb630/metadata/sequencing_protocol
e07897dc-c25e-453d-b04c-27ca33adb630/metadata/specimen_from_organism

e579d0f4-18ef-4d42-90fa-4d2afb60a862/descriptor
e579d0f4-18ef-4d42-90fa-4d2afb60a862/links
e579d0f4-18ef-4d42-90fa-4d2afb60a862/metadata/cell_suspension
e579d0f4-18ef-4d42-90fa-4d2afb60a862/metadata/collection_protocol
e579d0f4-18ef-4d42-90fa-4d2afb60a862/metadata/dissociation_protocol
e579d0f4-18ef-4d42-90fa-4d2afb60a862/metadata/donor_organism
e579d0f4-18ef-4d42-90fa-4d2afb60a862/metadata/enrichment_protocol
e579d0f4-18ef-4d42-90fa-4d2afb60a862/metadata/library_preparation_protocol
e579d0f4-18ef-4d42-90fa-4d2afb60a862/metadata/process
e579d0f4-18ef-4d42-90fa-4d2afb60a862/metadata/project
e579d0f4-18ef-4d42-90fa-4d2afb60a862/metadata/sequence_file
e579d0f4-18ef-4d42-90fa-4d2afb60a862/metadata/sequencing_protocol
e579d0f4-18ef-4d42-90fa-4d2afb60a862/metadata/specimen_from_organism
e579d0f4-18ef-4d42-90fa-4d2afb60a862/metadata/supplementary_file

e5d45579-1f5b-48c3-b568-320d93e7ca72/links
e5d45579-1f5b-48c3-b568-320d93e7ca72/metadata/project

e8808cc8-4ca0-4096-80f2-bba73600cba6/links
e8808cc8-4ca0-4096-80f2-bba73600cba6/metadata/project

eaefa1b6-dae1-4414-953b-17b0427d061e/links
eaefa1b6-dae1-4414-953b-17b0427d061e/metadata/project

efea6426-510a-4b60-9a19-277e52bfa815/links
efea6426-510a-4b60-9a19-277e52bfa815/metadata/project
```

We need to verify if these changes are intended and if we do want to proceed in sending them to DCP along with the project metadata update

**Comment by gabsie** [comment]
1. Some holding off until Amnon reviews a PR and Hannes do some checks for the data browser. 
2. A ticket to unblock DCP1 datasets

We also have a meeting today to understand things.

**Comment by aaclan-ebi** [comment]
Hannes confirmed that there's no need to wait for Data Browser changes. https://embl-ebi-ait.slack.com/archives/C9XD6L0AD/p1633970374034200?thread_ts=1633948217.032100&cid=C9XD6L0AD

Daniel confirmed as well there's a way to check the project estimated cell count via Azul in dev.

2 things to make sure as discussed with @gabsie(virtual call) and @amnonkhen (offline/f2f) last Friday:
1. We're not submitting projects which shouldn't be submitted yet (3 projects identified so far)
2. We're not overriding the changes done directly to the staging area before or doing any unintended changes. I need to verify the projects I noted here: https://app.zenhub.com/workspaces/dcp-ingest-product-development-5f71ca62a3cb47326bdc1b5c/issues/ebi-ait/dcp-ingest-central/445#issuecomment-936775977

**Comment by aaclan-ebi** [comment]
I created a ticket to spin out the work for DCP1 datasets as advised by @amnonkhen . This ticket will now only cover the DCP2 projects.

**Comment by aaclan-ebi** [comment]
Most of the unintended changes are changes in the links.json filenames:

e.g.
```
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/559bb888-7829-41f2-ace5-2c05c7eb81e9/links/:
      1711  2021-10-01T18:51:02Z  gs://broad-dsp-monster-hca-prod-ebi-storage/prod/559bb888-7829-41f2-ace5-2c05c7eb81e9/links/00e70c76-2030-4063-a83a-9885921a7dcd_2020-12-14T11:35:31.183000Z_559bb888-7829-41f2-ace5-2c05c7eb81e9.json
      1711  2021-02-18T21:29:56Z  gs://broad-dsp-monster-hca-prod-ebi-storage/prod/559bb888-7829-41f2-ace5-2c05c7eb81e9/links/00e70c76-2030-4063-a83a-9885921a7dcd_2020-12-14T11:35:31.183Z_559bb888-7829-41f2-ace5-2c05c7eb81e9.json

      1711  2021-10-01T18:51:12Z  gs://broad-dsp-monster-hca-prod-ebi-storage/prod/559bb888-7829-41f2-ace5-2c05c7eb81e9/links/0705bb56-818a-4b2c-a6e5-00e0d523d73c_2020-12-14T11:35:31.174000Z_559bb888-7829-41f2-ace5-2c05c7eb81e9.json
      1711  2021-02-18T21:29:55Z  gs://broad-dsp-monster-hca-prod-ebi-storage/prod/559bb888-7829-41f2-ace5-2c05c7eb81e9/links/0705bb56-818a-4b2c-a6e5-00e0d523d73c_2020-12-14T11:35:31.174Z_559bb888-7829-41f2-ace5-2c05c7eb81e9.json
```

It seems that a lot of projects' links.json files were first exported with 3 digit precision in seconds.

It looks like the DCP2 specs explicitly specified that there should be 6 digit precision in the seconds part of the version.
 https://github.com/HumanCellAtlas/dcp2/blob/0c38450ce714b5b6f0937c7c06a199ce0130c916/docs/dcp2_system_design.rst#object-naming

The change in the exporter to use 6 digit precision was pushed last **Oct 2020** https://github.com/ebi-ait/ingest-exporter/commit/7c558012a428f1e9965c2da314c01ce6a4c630c0. It seems that all projects submitted before this has links.json with less than 6 digit precision in the seconds in their versions. It seems that they have been exported without any issues. 

There are 51 projects which have links.json files with less than 6 precision. I believe it's okay to proceed to import these projects.

**Comment by aaclan-ebi** [comment]
Other unintended changes:

453d7ee2-319f-496c-9862-99d397870b63 - not eligible
783c9952-a4ae-4106-a6ce-56f20ce27f88 - rachel submitted this, project is already using 15.0.0 - this is false positive in the check i did
7adede6a-0ab7-45e6-9b67-ffe7466bec1f - failed in patching, should be retried and re-export
a815c84b-8999-433f-958e-422c0720e00d - this project is stalled 
e07897dc-c25e-453d-b04c-27ca33adb630 - this is a duplicate and was already recently deleted in Ingest and staging area
e579d0f4-18ef-4d42-90fa-4d2afb60a862 - this project is stalled

@ESapenaVentura there's a project which has lib prep protocol update. Is this intended? Do we have this in a different ticket? We could proceed with importing them to Terra at the same time as the project metadata upgrade to 15.0.0.

2086eb05-10b9-432b-b7f0-169ccc49d270 - has update for lib prep protocol
```
$ gsutil cat gs://broad-dsp-monster-hca-prod-ebi-storage/prod/2086eb05-10b9-432b-b7f0-169ccc49d270/metadata/library_preparation_protocol/ff7925dc-34f2-4e0b-a56b-a5b2f8ad68ef_2020-08-04T10:39:04.787000Z.json | jq > 1.json
$ gsutil gsutil cat gs://broad-dsp-monster-hca-prod-ebi-storage/prod/2086eb05-10b9-432b-b7f0-169ccc49d270/metadata/library_preparation_protocol/ff7925dc-34f2-4e0b-a56b-a5b2f8ad68ef_2021-09-15T14:30:16.128000Z.json | jq > 2.json

$ diff 1.json 2.json
21c21
<     "text": "10X 3' v2 sequencing",
---
>     "text": "10x 3' v2",
23c23
<     "ontology_label": "10X 3' v2 sequencing"
---
>     "ontology_label": "10x 3' v2"
36c36
<     "update_date": "2020-08-04T10:39:11.187Z",
---
>     "update_date": "2021-09-15T14:30:20.551Z",
```

The ontology text and label were updated from  `10X 3' v2 sequencing` to `10X 3' v2`

**Comment by ESapenaVentura** [comment]
@aaclan-ebi and I spoke about this on slack!

**Comment by ESapenaVentura** [comment]
Import team missed the import of a project - Waiting to see if they are happy to import.

**Comment by aaclan-ebi** [comment]
Awaiting Data Import team's successful import of the [58 projects](https://embl-ebi-ait.slack.com/archives/C01360XN04S/p1634319892222500?thread_ts=1631635383.071300&cid=C01360XN04S
) https://embl-ebi-ait.slack.com/archives/C01360XN04S/p1634834548001300?thread_ts=1631635383.071300&cid=C01360XN04S

**Comment by aaclan-ebi** [comment]
Reexported 2 submissions today:

1. 7adede6a-0ab7-45e6-9b67-ffe7466bec1f
https://contribute.data.humancellatlas.org/projects/detail?uuid=7adede6a-0ab7-45e6-9b67-ffe7466bec1f
https://contribute.data.humancellatlas.org/submissions/detail?uuid=e20e2ab7-3e89-4f9f-8ce2-91b2e1866a78&project=7adede6a-0ab7-45e6-9b67-ffe7466bec1f

2. ccef38d7-aa92-4010-9621-c4c7b1182647
https://contribute.data.humancellatlas.org/projects/detail?uuid=ccef38d7-aa92-4010-9621-c4c7b1182647
https://contribute.data.humancellatlas.org/submissions/detail?uuid=cdb6b0dd-73f2-49ea-96ad-2512e29624b2&project=ccef38d7-aa92-4010-9621-c4c7b1182647


Added **53 more projects** for importing:

```
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/24c654a5-caa5-440a-8f02-582921f2db4a
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/842605c7-375a-47c5-9e2c-a71c2c00fcad
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/2a64db43-1b55-4639-aabb-8dba0145689d
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/a96b71c0-78a7-42d1-88ce-83c78925cfeb
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/2086eb05-10b9-432b-b7f0-169ccc49d270
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/0d4b87ea-6e9e-4569-82e4-1343e0e3259f
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/41fb1734-a121-4616-95c7-3b732c9433c7
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/63b5b6c1-bbcd-487d-8c2e-0095150c1ecd
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/955dfc2c-a8c6-4d04-aa4d-907610545d11
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/559bb888-7829-41f2-ace5-2c05c7eb81e9
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/1c6a960d-52ac-44ea-b728-a59c7ab9dc8e
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/6c040a93-8cf8-4fd5-98de-2297eb07e9f6
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/07073c12-8006-4710-a00b-23abdb814904
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/e5d45579-1f5b-48c3-b568-320d93e7ca72
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/53c53cd4-8127-4e12-bc7f-8fe1610a715c
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/1cd1f41f-f81a-486b-a05b-66ec60f81dcf
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/996120f9-e84f-409f-a01e-732ab58ca8b9
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/efea6426-510a-4b60-9a19-277e52bfa815
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/0c09fade-e079-4fde-8e60-6725b8c1d84b
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/ce7b12ba-664f-4f79-8fc7-3de6b1892183
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/0fd8f918-62d6-4b8b-ac35-4c53dd601f71
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/d3a4ceac-4d66-4984-9704-2570c0647a56
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/1ad6cee6-53a5-4804-a870-e5bd991186e0
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/16ed4ad8-7319-46b2-8859-6fe1c1d73a82
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/a39728aa-70a0-4201-b0a2-81b7badf3e71
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/4bec484d-ca7a-47b4-8d48-8830e06ad6db
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/4037007b-0eff-4e6d-b7bd-8dd8eec80143
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/bd400331-54b9-4fcc-bff6-6bb8b079ee1f
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/3089d311-f9ed-44dd-bb10-397059bad4dc
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/c1810dbc-16d2-45c3-b45e-3e675f88d87b
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/52d10a60-c8d1-4d06-8a5e-af0d5c0d5034
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/3a694703-3084-4ece-9abe-d935fd5f6748
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/ce33dde2-382d-448c-b6ac-bfb424644f23
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/946c5add-47d1-402a-97bb-a5af97e8bce7
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/0c3b7785-f74d-4091-8616-a68757e4c2a8
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/05657a59-9f9d-4bb9-b77b-24be13aa5cea
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/ccd1f1ba-74ce-469b-9fc9-f6faea623358
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/b51f49b4-0d2e-4cbd-bbd5-04cd171fc2fa
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/eaefa1b6-dae1-4414-953b-17b0427d061e
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/6072616c-8794-4b20-8f52-fb15992ea5a4
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/1ce3b3dc-02f2-44a8-96da-d6d107b27a76
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/52b29aa4-c8d6-42b4-807a-b35be94469ca
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/c6ad8f9b-d26a-4811-b2ba-93d487978446
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/520afa10-f9d2-4e93-ab7a-26c4c863ce18
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/d2111fac-3fc4-4f42-9b6d-32cd6a828267
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/e8808cc8-4ca0-4096-80f2-bba73600cba6
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/05be4f37-4506-429b-b112-506444507d62
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/03c6fce7-789e-4e78-a27a-664d562bb738
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/23587fb3-1a4a-4f58-ad74-cc9a4cb4c254
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/2a72a4e5-66b2-405a-bb7c-1e463e8febb0
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/248fcf03-16c6-4a41-b6cc-aad4d894ca42
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/ccef38d7-aa92-4010-9621-c4c7b1182647
gs://broad-dsp-monster-hca-prod-ebi-storage/prod/7adede6a-0ab7-45e6-9b67-ffe7466bec1f
```

https://embl-ebi-ait.slack.com/archives/C01360XN04S/p1635177648009500?thread_ts=1631635383.071300&cid=C01360XN04S

**Comment by gabsie** [comment]
@aaclan-ebi is going to check if these projects are in the release

**Comment by aaclan-ebi** [comment]
Confirmed that the 111 projects will be included in Release 11.

**Comment by ipediez** [comment]
Two different entities pointing to the same file. @aaclan-ebi tried to push the metadata, @ESapenaVentura to take a look at this issue: investigate the reason, decide which entities to delete

**Comment by aaclan-ebi** [comment]
Confirmed 109 out of 111 project cell counts look good in the DCP release 11 
https://embl-ebi-ait.slack.com/archives/C014HMTV0FN/p1636135537064500?thread_ts=1636134761.064400&cid=C014HMTV0FN

Note 1 project was excluded as it has duplicate submission in Ingest https://embl-ebi-ait.slack.com/archives/C9XD6L0AD/p1635423333013700?thread_ts=1635211460.009200&cid=C9XD6L0AD 
(insert ticket in ops board  created to fix it)

1 project seems to be missing and have been removed both in Data Browser and Ingest.

I used Data Browser's API to get the project data: `https://service.azul.data.humancellatlas.org/index/projects/{project_uuid}?catalog=dcp11`
